test/AWS.Deploy.CLI.IntegrationTests/ServerMode/GetApplyOptionSettings.cs

```diff
@@ -53,24 +57,20 @@ public GetApplyOptionSettings()
             _testAppManager = new TestAppManager();
         }
 
-        public Task<AWSCredentials> ResolveCredentials()
+        public TemplateMetadataReader GetTemplateMetadataReader(string templateBody)
         {
-            var testCredentials = FallbackCredentialsFactory.GetCredentials();
-            return Task.FromResult<AWSCredentials>(testCredentials);
+            var templateMetadataReader = new TemplateMetadataReader(_mockAWSClientFactory.Object);
+            var cfResponse = new GetTemplateResponse();
+            cfResponse.TemplateBody = templateBody;
+            _mockAWSClientFactory.Setup(x => x.GetAWSClient<IAmazonCloudFormation>(It.IsAny<string>())).Returns(_mockCFClient.Object);
+            _mockCFClient.Setup(x => x.GetTemplateAsync(It.IsAny<GetTemplateRequest>(), It.IsAny<CancellationToken>())).ReturnsAsync(cfResponse);
+            return templateMetadataReader;
         }
 
-        private async Task<DeploymentStatus> WaitForDeployment(RestAPIClient restApiClient, string sessionId)
+        public Task<AWSCredentials> ResolveCredentials()
         {
-            // Do an initial delay to avoid a race condition of the status being checked before the deployment has kicked off.
-            await Task.Delay(TimeSpan.FromSeconds(3));
-
-            await WaitUntilHelper.WaitUntil(async () =>
-            {
-                DeploymentStatus status = (await restApiClient.GetDeploymentStatusAsync(sessionId)).Status; ;
-                return status != DeploymentStatus.Executing;
-            }, TimeSpan.FromSeconds(1), TimeSpan.FromMinutes(15));
-
-            return (await restApiClient.GetDeploymentStatusAsync(sessionId)).Status;
+            var testCredentials = FallbackCredentialsFactory.GetCredentials();
```

Comment:
Do we still need to access the user's default credentials if we're not actually deploying anything?